app/controllers/nodes_controller.rb

```diff
@@ -89,6 +89,181 @@ class NodesController < AuthenticatedUsersController
+    #       "journal_size": 5192,
+    #       "journal_colocation": "false",
+    #       "storage_disks": [
+    #         {
+    #           "device": "/dev/sda",
+    #           "journal": "/dev/sdc"
+    #         },
+    #         {
+    #           "device": "/dev/sdb",
+    #           "journal": "/dev/sdc"
+    #         }
+    #       ]
+    #     },
+    #     "10.0.0.30": {
+    #       "role": "ceph/osd",
+    #       "provisioning_ip": "10.0.0.30",
+    #       "journal_colocation": "true",
+    #       "storage_disks": [
+    #         {
+    #           "device": "/dev/sda"
+    #         },
+    #         {
+    #           "device": "/dev/sdb"
+    #         }
+    #       ]
+    #     }
+    #   }
+    # }
+    #
+    # Job structure:
+    #
+    # {
+    #   "integration_id": "9a4b84e0-17b3-4543-af9f-e42000c52bfc",
+    #   "run": "tendrl.flows.CreateCluster",
+    #   "status": "new",
+    #   "type": "node",
+    #   "node_ids": [],
+    #   "tags": ["provisioner/ceph"],
+    #   "parameters": {
+    #     "sds_type": "ceph",
+    #     "sds_version": "10.2.5",
+    #     "name": "MyCluster",
+    #     "TendrlContext.integration_id": "9a4b84e0-17b3-4543-af9f-e42000c52bfc",
+    #     "Node[]": [
+    #       "3a95fd96-876d-439a-a64d-70332c069aaa",
+    #       "3943fab1-9ed2-4eb6-8121-5a69499c4568",
+    #       "b10e00e9-e444-41c2-9517-df2118b42731"
+    #     ],
+    #     "fsid": "140cd3d5-58e4-4935-a954-d946ceff371d",
+    #     "public_network": "192.168.128.0/24",
+    #     "cluster_network": "192.168.220.0/24",
+    #     "ceph_conf_overrides": {
+    #       "global": {
+    #         "osd_pool_default_pg_num": 128,
+    #         "pool_default_pgp_num": 1
+    #       }
+    #     },
+    #     "node_configuration": {
+    #       "3a95fd96-876d-439a-a64d-70332c069aaa": {
+    #         "role": "ceph/mon",
+    #         "provisioning_ip": "10.0.0.24",
+    #         "monitor_interface": "eth0"
+    #       },
+    #       "3943fab1-9ed2-4eb6-8121-5a69499c4568": {
+    #         "role": "ceph/osd",
+    #         "provisioning_ip": "10.0.0.29",
+    #         "journal_size": 5192,
+    #         "journal_colocation": "false",
+    #         "storage_disks": [
+    #           {
+    #             "device": "/dev/sda",
+    #             "journal": "/dev/sdc"
+    #           },
+    #           {
+    #             "device": "/dev/sdb",
+    #             "journal": "/dev/sdc"
+    #           }
+    #         ]
+    #       },
+    #       "b10e00e9-e444-41c2-9517-df2118b42731": {
+    #         "role": "ceph/osd",
+    #         "provisioning_ip": "10.0.0.30",
+    #         "journal_colocation": "true",
+    #         "storage_disks": [
+    #           {
+    #             "device": "/dev/sda"
+    #           },
+    #           {
+    #             "device": "/dev/sdb"
+    #           }
+    #         ]
+    #       }
+    #     }
+    #   }
+    # }
+
+    missing_params = []
```

Comment:
Add the validation in something like cluster_validator.rb like we have for user_validator.rb, you need to look into active_model for that.